lighthouse-core/gather/driver.js

```diff
@@ -570,9 +570,14 @@ class Driver {
       });
   }
 
-  beginTrace() {
+  /**
+   * @param {{traceCategories: string=}=} flags
+   */
+  beginTrace(flags) {
+    const additionalCategories = flags && flags.traceCategories && flags.traceCategories.split(',');
+    const traceCategories = new Set(this._traceCategories.concat(additionalCategories || []));
```

Comment:
let's just drop in `const _uniq = arr => Array.from(new Set(arr));` and use it so we're not tossing this type from Set back to array a few lines later.